src/file_object.cpp

```diff
@@ -404,4 +404,9 @@ Value FileObject::chmod(Env *env, Value mode, Value path) {
     return Value::integer(1); // return # of files
 }
 
+Value FileObject::stat(Env *env, Value path) {
+    path = fileutil::convert_using_to_path(env, path);
+    return new FileStatObject { path->as_string()->c_str() };
```

Comment:
It gets used here, though it is kind of stubbed out at this point.

Reply:
Oh oK I'll wait till you're done. 😁

Reply:
This is my first foray into adding a new class and was mainly looking for early feedback on this draft to let me know if I really did things the hard way here - before I got too far along - or any other advice you might have.  It's still really unpolished/unfinished.